src/MDOP.jl

```diff
@@ -4,16 +4,150 @@ using Distances
+  of false nearest neighbors, in order to the let algorithm terminate and stop
+  the embedding procedure (`0 <= fnn_thres < 1).
+* `r::Real = 2`: The threshold for the tolerable relative increase of the distance
+  between the nearest neighbors, when increasing the embedding dimension.
+* `β_condition::Bool = false`: When set `true`, the function also returnes the `β`-statistics
+  of all embedding cycles.
+
+## Description
+The method works iteratively and gradually builds the final embedding vectors
+`Y`. Based on the [`beta_statistic`](@ref) the algorithm picks an optimal delay
+value `τ` for each embedding cycle as the global maximum of `β`. In case of
+multivariate embedding, i.e. when embedding a set of time series (`s::Dataset`),
+the optimal delay value `τ` is chosen as the maximum from all maxima's of all
+considered `β`-statistics for each embedding cycle. The range of
+considered delay values is determined in `τs` and for the nearest neighbor
+search we respect the Theiler window `w`. After each embedding cycle the FNN-
+statistic `FNNs` [^Hegger1999][^Kennel1992] is being checked and as soon as this statistic
+drops below the threshold `fnn_thres`, the algorithm breaks. In order to
+increase the  practability of the method the algorithm also breaks, when the
+FNN-statistic `FNNs` increases . The final embedding vector is stored in `Y`
+(`Dataset`). The chosen delay values for each embedding cycle are stored in the
+`τ_vals` and the according time series number chosen for the according delay
+value in `τ_vals` is stored in `ts_vals`. For univariate embedding (`s::Vector`)
+`ts_vals` is a vector of ones of length `τ_vals`, because there is simply just
+one `β`-statistic for each embedding cycle as an `Array` of `Vector`s.
 
 [^Nichkawde2013]: Nichkawde, Chetan (2013). [Optimal state-space reconstruction using derivatives on projected manifold. Physical Review E 87, 022905](https://doi.org/10.1103/PhysRevE.87.022905).
+[^Hegger1999]: Hegger, Rainer and Kantz, Holger (1999). [Improved false nearest neighbor method to detect determinism in time series data. Physical Review E 60, 4970](https://doi.org/10.1103/PhysRevE.60.4970).
+[^Kennel1992]: Kennel, M. B., Brown, R., Abarbanel, H. D. I. (1992). [Determining embedding dimension for state-space reconstruction using a geometrical construction. Phys. Rev. A 45, 3403] (https://doi.org/10.1103/PhysRevA.45.3403).
 """
-function MDOP(Y...)
+function MDOP(s::Vector{<:Real}; τs = 0:50 , w::Int = 1, fnn_thres::Real = 0.05,
+    r::Real = 2, β_condition::Bool=false)
+    @assert 0 <= fnn_thres < 1 "Please select a valid breaking criterion, i.e. a threshold value `fnn_thres` ∈ [0 1)"
+    @assert all(x -> x ≥ 0, τs)
+
+    T = eltype(s)
+
+    max_num_of_cycles = 50 # assure that the algorithm will break after 50 embedding cycles
+
+    metric = Euclidean()
+    # normalize input time series (especially important for fnn-computation)
+    s .= (s.-mean(s))./std(s)
```

Comment:
Wait, here we are altering the input timeseries of the user, without warning. I Didn't realize this before hand when I recommended `.=`. 

Please do `s = regularize(s)` instead. This leaves the original timeseries unchanged. If not, it can make user code behave very weirdly without them knowing why.